flocker/acceptance/endtoend/test_cli.py

```diff
@@ -0,0 +1,55 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for the ``flocker-deploy`` command line tool.
+"""
+
+from twisted.trial.unittest import TestCase
+
+from ..testtools import require_flocker_cli, require_cluster
+from ...testtools import random_name
+
+
+class FlockerDeployTests(TestCase):
+    """
+    Tests for ``flocker-deploy``.
+    """
+    @require_flocker_cli
+    @require_cluster(1)
+    def test_deploy(self, cluster):
+        """
+        Deploying an application to one node and not another puts the
+        application where expected.
+        """
+        [node_1] = cluster.nodes
+        name = random_name(self)
+
+        minimal_deployment = {
+            u"version": 1,
+            u"nodes": {
+                node_1.reported_hostname: [name],
+            },
+        }
+
+        minimal_application = {
+            u"version": 1,
+            u"applications": {
+                name: {
+                    # Our config format doesn't support command-lines, so
+                    # we can't use standard image and need to pick
+                    # something small that will not immediately exit.
+                    u"image": u"openshift/busybox-http-app",
```

Comment:
Yeah, ideally we'd use our own image. `build` won't work since we don't support it.

On the other hand we're killing `flocker-deploy` so I'd like to minimize effort spent on this particular test. So I'll just harcode a tag, yeah.